core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/devui/OpenWebUIJsonRPCService.java

```diff
@@ -0,0 +1,145 @@
+                        .getHostPortSpec();
+                return "http://localhost" + ":" + publicPort;
+            }
+        }
+        return null;
+    }
+
+    public InspectContainerResponse inspectOpenWebUIContainer() {
+        return DockerClientFactory.lazyClient().listContainersCmd().exec()
+                .stream()
+                .filter(OpenWebUIJsonRPCService::isOpenWebUIContainer)
+                .findFirst()
+                .map(c -> c.getId())
+                .map(id -> DockerClientFactory.lazyClient().inspectContainerCmd(id).exec())
+                .orElse(null);
+    }
+
+    public CreateContainerResponse startOpenWebUI(String image,
+            boolean requestGpu,
+            Map<Integer, Integer> portBindings,
+            Map<String, String> envVars,
+            Map<String, String> envVarMappings,
+            Map<String, String> volumes) {
+        if (!isOpenWebUIRunning()) {
+
+            List<PortBinding> allPortBindings = new ArrayList<>();
+            List<String> allEnvVars = new ArrayList<>();
+            List<Bind> allBinds = new ArrayList<>();
+            List<Volume> allVolumes = new ArrayList<>();
+            List<DeviceRequest> allDeviceRequests = new ArrayList<>();
+
+            for (var e : portBindings.entrySet()) {
+                allPortBindings.add(PortBinding.parse(e.getKey() + ":" + e.getValue()));
+            }
+
+            for (var e : envVars.entrySet()) {
+                allEnvVars.add(e.getKey() + "=" + e.getValue());
+            }
+
+            for (var e : envVarMappings.entrySet()) {
+                String key = e.getKey();
+                String value = getEnvVarValue(e.getValue());
+                if (value != null) {
+                    allEnvVars.add(key + "=" + value);
+                }
+            }
+            for (var e : volumes.entrySet()) {
+                String path = e.getKey();
+                Volume volume = new Volume(e.getValue());
+                allBinds.add(new Bind(path, volume));
+                allVolumes.add(volume);
+                try {
+                    DockerClientFactory.lazyClient().inspectVolumeCmd(path).exec();
+                } catch (NotFoundException nfe) {
+                    DockerClientFactory.lazyClient().createVolumeCmd().withName(path).exec();
+                }
+            }
+
+            if (requestGpu) {
+                DeviceRequest gpu = new DeviceRequest().withCount(-1).withCapabilities(List.of(List.of("gpu")));
+                allDeviceRequests.add(gpu);
+            }
+
+            HostConfig hostConfig = new HostConfig()
+                    .withBinds(allBinds)
+                    .withPortBindings(allPortBindings)
+                    .withExtraHosts("host.docker.internal:host-gateway")
+                    .withDeviceRequests(allDeviceRequests);
+
+            CreateContainerResponse container = DockerClientFactory.lazyClient()
+                    .createContainerCmd(image)
```

Comment:
I had errors that the image doesn't exist

```
 com.github.dockerjava.api.exception.NotFoundException: Status 404: {"cause":"image not known","message":"no such image: ghcr.io/open-webui/open-webui:main: image not known","response":404}
```

and had to manually run `podman pull ghcr.io/open-webui/open-webui:main` to get it working. Could it be made to pull the image automatically?

Reply:
I'll look into that.